MonoGame.Framework/Random.cs

```diff
@@ -0,0 +1,404 @@
+
+            #region Floating Point Next methods
+
+            #region Double Next methods
+
+            /// <summary>
+            /// Returns a non-negative random double-precision floating point number.
+            /// </summary>
+            /// <returns>Random double-precision floating point number</returns>
+            public double NextDouble()
+            {
+                return Remap(UInt64ToDouble(XORShift64Star()), double.MinValue, double.MaxValue, 0, double.MaxValue);
+            }
+
+            /// <summary>
+            /// Returns a non-negative random double-precision floating point number that is less than the specified maximum.
+            /// </summary>
+            /// <param name="max">Maximum possible value</param>
+            /// <returns>Random double-precision floating point number</returns>
+            public double NextDouble(double max)
+            {
+                return Remap(UInt64ToDouble(XORShift64Star()), double.MinValue, double.MaxValue, 0, max);
+            }
+
+            /// <summary>
+            /// Returns a random double-precision floating point number that is within a specified range.
+            /// </summary>
+            /// <param name="min">Minimum possible value</param>
+            /// <param name="max">Maximum possible value</param>
+            /// <returns>Random double-precision floating point number</returns>
+            public double NextDouble(double min, double max)
+            {
+                return Remap(UInt64ToDouble(XORShift64Star()), double.MinValue, double.MaxValue, min, max);
+            }
+
+            #endregion
+
+            #region Float Next methods
+
+            /// <summary>
+            /// Returns a non-negative random single-precision floating point number.
+            /// </summary>
+            /// <returns>Random single-precision floating point number</returns>
+            public float NextFloat()
+            {
+                return (float)NextDouble();
+            }
+
+            /// <summary>
+            /// Returns a non-negative random single-precision floating point number that is less than the specified maximum.
+            /// </summary>
+            /// <param name="max">Maximum possible value</param>
+            /// <returns>Random single-precision floating point number</returns>
+            public float NextFloat(float max)
+            {
+                return (float)NextDouble((double)max);
+            }
+
+            /// <summary>
+            /// Returns a random single-precision floating point number that is within a specified range.
+            /// </summary>
+            /// <param name="min">Minimum possible value</param>
+            /// <param name="max">Maximum possible value</param>
+            /// <returns>Random single-precision floating point number</returns>
+            public float NextFloat(float min, float max)
+            {
+                return (float)NextDouble((double)min, (double)max);
```

Comment:
Float is more common than double, especially in MonoGame and it deserves their own implementation.. @tomspilman ? Remap with float overload could be more good than raw conversion from double ?